shared/src/commonMain/kotlin/org/jetbrains/kotlinconf/screens/Speakers.kt

```diff
@@ -1,24 +1,101 @@
+import org.jetbrains.kotlinconf.Speakers
+import org.jetbrains.kotlinconf.ui.components.Divider
+import org.jetbrains.kotlinconf.ui.components.MainHeaderContainer
+import org.jetbrains.kotlinconf.ui.components.MainHeaderContainerState
+import org.jetbrains.kotlinconf.ui.components.MainHeaderSearchBar
+import org.jetbrains.kotlinconf.ui.components.MainHeaderTitleBar
+import org.jetbrains.kotlinconf.ui.components.SpeakerCard
+import org.jetbrains.kotlinconf.ui.components.TopMenuButton
+import org.jetbrains.kotlinconf.ui.theme.KotlinConfTheme
+import org.jetbrains.kotlinconf.utils.containsDiacritics
+import org.jetbrains.kotlinconf.utils.removeDiacritics
+import kotlinconfapp.ui_components.generated.resources.Res as UiRes
 
 @Composable
 fun Speakers(
-    onBack: () -> Unit,
+    speakers: Speakers,
+    onSpeaker: (SpeakerId) -> Unit,
 ) {
-    Column {
-        Image(
-            painterResource(Res.drawable.arrow_left_24),
-            "back",
-            modifier = Modifier.clickable { onBack() })
-        StyledText("Speakers placeholder")
+    var searchState by remember { mutableStateOf(MainHeaderContainerState.Title) }
+    var searchText by remember { mutableStateOf("") }
+
+    Column(Modifier.fillMaxSize()) {
+        MainHeaderContainer(
+            state = searchState,
+            titleContent = {
+                MainHeaderTitleBar(
+                    title = stringResource(Res.string.speakers_title),
+                    endContent = {
+                        TopMenuButton(
+                            icon = UiRes.drawable.search_24,
+                            onClick = { searchState = MainHeaderContainerState.Search },
+                            contentDescription = stringResource(UiRes.string.main_header_search_hint)
+                        )
+                    }
+                )
+            },
+            searchContent = {
+                MainHeaderSearchBar(
+                    searchValue = searchText,
+                    onSearchValueChange = { searchText = it },
+                    onClose = {
+                        searchState = MainHeaderContainerState.Title
+                        searchText = ""
+                    },
+                    onClear = { searchText = "" }
+                )
+            }
+        )
+
+        Divider(1.dp, KotlinConfTheme.colors.strokePale)
+
+        val filtered = speakers.all.filter {
+            // Look for exact matches if diacritics are present, ignore all diacritics otherwise
+            val diacriticsSearch = searchText.containsDiacritics()
+            val targetName = if (diacriticsSearch) it.name else it.name.removeDiacritics()
+            val targetPosition = if (diacriticsSearch) it.position else it.position.removeDiacritics()
+            val searchPattern = searchText.toRegex(RegexOption.IGNORE_CASE)
+
+            searchText.isEmpty() || searchPattern.containsMatchIn(targetName) || searchPattern.containsMatchIn(targetPosition)
```

Comment:
Nit: if `searchText.isEmpty()` I would just skip this entire `filter` operation to be slightly cleaner